### PR TITLE
Add more NodeJS versions to test against

### PR DESCRIPTION
**Pull request by JakeChampion**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,31 @@
-sudo: false
-language: node_js
 node_js:
-  - "0.10"
+  - "v5.1"
```
Comment (ljharb) [inline .travis.yml]: 5.2 and 5.3?

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,31 @@
-sudo: false
-language: node_js
```
Comment (ljharb) [inline .travis.yml]: this is required, or else it will use the wrong travis image

**Comment by ljharb** [comment]
Could you use https://github.com/ljharb/is-callable/blob/master/.travis.yml as a baseline instead? it handles older node versions quite well, and i use it across all of my modules

**Comment by JakeChampion** [comment]
@ljharb I've done just that. May I ask why so many versions are marked as being allowed to fail?

**Comment by ljharb** [comment]
@JakeChampion it's useful to test on as many versions as possible, but typically it doesn't matter if it fails on "not the latest in the minor version line".

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,46 @@
...
+  - "0.10"
+  - "0.9"
+  - "0.8"
+  - "0.6"
+  - "0.4"
+before_install:
+  - 'if [ "${TRAVIS_NODE_VERSION}" != "0.9" ]; then case "$(npm --version)" in 1.*) npm install -g npm@1.4.28 ;; 2.*) npm install -g npm@2 ;; esac ; fi'
+  - 'if [ "${TRAVIS_NODE_VERSION}" != "0.6" ] && [ "${TRAVIS_NODE_VERSION}" != "0.9" ]; then npm install -g npm; fi'
+script:
+  - 'if [ "${TRAVIS_NODE_VERSION}" != "4.2" ]; then npm run tests-only ; else npm test ; fi'
```
Comment (ljharb) [inline .travis.yml]: only catch here: `npm run tests-only` needs to exist in package.json scripts - in this case, there's no linter (yet), so `npm test` [here](https://github.com/mathiasbynens/Array.of/blob/master/package.json#L29) can be renamed to "tests-only", and `test` set to `npm run tests-only` for now.

File: .travis.yml [hunk 4]
```diff
@@ -1,6 +1,46 @@
...
+sudo: false
+matrix:
+  fast_finish: true
+  allow_failures:
+    - node_js: "0.6"
+    - node_js: "0.4"
```
Comment (ljharb) [inline .travis.yml]: all files everywhere should end in a trailing newline please :-)

**Comment by JakeChampion** [comment]
@ljharb Comments have been addressed. :+1:

**Pull request merged**